### PR TITLE
feat: remove `try-with`

### PR DESCRIPTION
**Pull request by jaschdoc**
Depends on https://github.com/flix/flix/pull/9409
Depends on https://github.com/flix/flix/pull/9398
Closes https://github.com/flix/flix/issues/9214

### CODE REVIEW

**Comment by jaschdoc** [comment]
@JonathanStarup we are removing support for `try-with`

File: main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala [hunk 1]
```diff
@@ -19,17 +19,17 @@ import ca.uwaterloo.flix.language.CompilationMessage
 import ca.uwaterloo.flix.language.ast.shared.Source
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala:19]: Can we avoid the reformatting?

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala [hunk 2]
```diff
@@ -1724,33 +1725,47 @@ object Weeder2 {
       }
     }
 
+    private def visitRunExpr(tree: Tree)(implicit sctx: SharedContext): Validation[Expr, CompilationMessage] = {
+      expect(tree, TreeKind.Expr.Run)
+      val maybeWith = pickAll(TreeKind.Expr.TryWithBodyFragment, tree)
+      flatMapN(
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1731]: Does this mirror the structure of visitTryExpr?

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala [hunk 3]
```diff
@@ -1724,33 +1725,47 @@ object Weeder2 {
       }
     }
 
+    private def visitRunExpr(tree: Tree)(implicit sctx: SharedContext): Validation[Expr, CompilationMessage] = {
+      expect(tree, TreeKind.Expr.Run)
+      val maybeWith = pickAll(TreeKind.Expr.TryWithBodyFragment, tree)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1730]: Rename TryWithBodyFragment?

File: main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala (hunk 2 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala:1731]: Yes

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**